etl::type_list_is_unique compilation error

I get the following compilation error on v20.46.0:

```plain
[build] In file included from include/etl/function_traits.h:35,
[build]                  from include/etl/private/delegate_cpp11.h:55,
[build]                  from include/etl/delegate.h:37,
[build]                  from include/etl/callback_timer_locked.h:35,
[build]                  from my source:
[build] include/etl/type_list.h:656:31: error: 'is_same' is not a member of 'std'; did you mean 'etl::is_same'? [-Wtemplate-body]
[build]   656 |     : etl::bool_constant<std::is_same<TTypeList, typename type_list_unique<TTypeList>::type>::value>
[build]       |                               ^~~~~~~
```

Compiler:

    c++.exe (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders, r5) 15.2.0

Relevant options:

    -DETL_NO_INITIALIZER_LIST -DETL_NO_STL -DETL_TARGET_OS_CMSIS_OS2 -std=gnu++20

May be related to #1307

--- a/include/etl/type_list.h
+++ b/include/etl/type_list.h
@@ -653,7 +653,7 @@ namespace etl
   struct type_list_is_unique
     // Create a unique version of the type list, and check if it is the same as the original list.
     // If they are the same, then all types in the original list are unique.
-    : etl::bool_constant<std::is_same<TTypeList, typename type_list_unique<TTypeList>::type>::value>
+    : etl::bool_constant<etl::is_same<TTypeList, typename type_list_unique<TTypeList>::type>::value>
   {
     ETL_STATIC_ASSERT((etl::is_type_list<TTypeList>::value), "TTypeList must be an etl::type_list");
   };

